### PR TITLE
Remove any

### PR DESCRIPTION
**Pull request by alunduil**

### CODE REVIEW

File: zfs/replicate/snapshot/type.py [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 
-class Snapshot(NamedTuple):
+@dataclass
+class Snapshot:
     """ZFS Snapshot Type."""
 
     filesystem: FileSystem
     name: str
-    previous: Optional[Any]  # mypy doesn't support nested types yet
+    previous: Optional["Snapshot"]  # mypy doesn't support nested types yet
```
Comment (Flameeyes) [inline zfs/replicate/snapshot/type.py:15]: Outdated comment?

File: zfs/replicate/snapshot/type.py [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
     timestamp: int
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: object) -> bool:
```
Comment (Flameeyes) [inline zfs/replicate/snapshot/type.py:18]: Would it make sense for other to be `: "Snapshot"` too?

**Review by Flameeyes — APPROVED** [review]

File: zfs/replicate/snapshot/type.py (hunk 1 above)
Comment (alunduil) [reply zfs/replicate/snapshot/type.py:15]: Yep, I'll remove that real quick.

File: zfs/replicate/snapshot/type.py (hunk 2 above)
Comment (alunduil) [reply zfs/replicate/snapshot/type.py:18]: mypy or pep8 wants the inherited signature to match so object is the only correct answer here.

**Comment by alunduil** [comment]
@Flameeyes  this is also finally ready for review.  Once I get these couple landed. I'll try updates one more time and then cut a new release.

**Review by Flameeyes — APPROVED** [review]

**Pull request merged**